src/main/java/com/jcabi/maven/plugin/UnwovenClasses.java

```diff
@@ -0,0 +1,120 @@
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.jcabi.maven.plugin;
+
+import com.jcabi.log.Logger;
+import java.io.File;
+import java.io.IOException;
+import org.apache.commons.io.FileUtils;
+import org.apache.maven.plugin.MojoFailureException;
+
+/**
+ * Operations on the unwoven classes, like storing them in a separate
+ * location from the woven ones. Unwoven classes are classes which weren't
+ * yet weaved by the aspect weaver.
+ * @author Mihai Andronache (amihaiemil@gmail.com)
+ * @version $Id$
+ * @since 0.15
+ */
+public final class UnwovenClasses {
+    /**
+     * Directory where unwoven classes are saved.
+     */
+    private final transient File unwoven;
+
+    /**
+     * Output directory from where the classes are taken.
+     */
+    private final transient File classes;
+
+    /**
+     * Maven execution phase.
+     */
+    private final transient String phase;
+
+    /**
+     * Constructor.
+     * @param uwvn Dir where unwoven classes go
+     * @param cls Directory where the classes are found
+     * @param phs Maven execution phase
+     */
+    public UnwovenClasses(final File uwvn, final File cls,
+        final String phs) {
+        this.unwoven = uwvn;
+        this.classes = cls;
+        this.phase = phs;
+    }
+
+    /**
+     * Perform the copy. Unwoven classes go in <b>unwoven</b> directory, while
+     * unwoven test classes go in <b>unwoven</b> + -test directory.
+     * @throws MojoFailureException If there is an IOException when
+     *  copying the files
+     */
+    public void copy() throws MojoFailureException {
+        if ("process-classes".equals(this.phase)) {
+            this.unwoven.mkdirs();
+            Logger.info(
+                this, "Unwoven classes will be copied to %s",
+                this.unwoven
+            );
+            this.copyContents(this.classes, this.unwoven);
+        } else if ("process-test-classes".equals(this.phase)) {
+            final String suffix = "-test";
+            final File unwovenTests = new File(
+                this.unwoven.getPath().concat(suffix)
+            );
+            unwovenTests.mkdirs();
+            Logger.info(
+                this, "Unwoven test classes will be copied to %s",
+                unwovenTests
+            );
+            this.copyContents(this.classes, unwovenTests);
+        }
+    }
+
+    /**
+     * Copies contents from one dir to another.
+     * @param from From directory
+     * @param dest Destination directory
+     * @throws MojoFailureException If something goes wrong while copying files
+     */
+    private void copyContents(
+        final File from, final File dest) throws MojoFailureException {
+        try {
+            FileUtils.cleanDirectory(dest);
+            FileUtils.copyDirectory(from, dest, false);
+        } catch (final IOException ex) {
+            throw new MojoFailureException(ex.getMessage(), ex);
```

Comment:
@amihaiemil actually we should provide some additional info here, see http://www.yegor256.com/2015/12/01/rethrow-exceptions.html